### ISSUE TITLE
[Build Error-Ubuntu 21.04] Packages not being linked and build failing at the last step(sudo make install)

### ISSUE THREAD

**Issue by Kashyapdevesh** [mlpack/mlpack#3079]
Hello everyone,
I'm following the linux build process listed at https://www.mlpack.org/doc/stable/doxygen/build.html and I'm facing a library linkage issue in the last step(sudo make install).

Initially, mlpack was not recognizing the libraries installed such as Rcpp,roxygen2 , etc and was showing the error shown below:

```
[  0%] Updating arma_config.hpp (if necessary)
[  0%] Built target mlpack_arma_config
Moving header files to include/mlpack/
[  0%] Built target mlpack_headers
[  1%] Generating mlpack.pc (pkg-config) file.
[  1%] Built target pkgconfig
[  6%] Built target mlpack
[  6%] Built target mlpack_julia_util
[  7%] Built target mlpack_julia_sparse_coding
[  7%] Built target generate_jl_test_julia_binding
[  7%] Built target mlpack_julia_test_julia_binding
[  7%] Built target generate_jl_adaboost
[  7%] Built target mlpack_julia_adaboost
[  7%] Built target generate_jl_approx_kfn
[  7%] Built target mlpack_julia_approx_kfn
[  8%] Built target generate_jl_bayesian_linear_regression
[  9%] Built target mlpack_julia_bayesian_linear_regression
[  9%] Built target generate_jl_cf
[  9%] Built target mlpack_julia_cf
[ 10%] Built target generate_jl_dbscan
[ 10%] Built target mlpack_julia_dbscan
[ 10%] Built target generate_jl_decision_stump
[ 11%] Built target mlpack_julia_decision_stump
[ 12%] Built target generate_jl_decision_tree
[ 12%] Built target mlpack_julia_decision_tree
[ 12%] Built target generate_jl_det
[ 12%] Built target mlpack_julia_det
[ 12%] Built target generate_jl_emst
[ 12%] Built target mlpack_julia_emst
[ 13%] Built target generate_jl_fastmks
[ 14%] Built target mlpack_julia_fastmks
[ 15%] Built target mlpack_julia_gmm_train
[ 15%] Built target generate_jl_gmm_train
[ 15%] Built target generate_jl_gmm_generate
[ 15%] Built target mlpack_julia_gmm_probability
[ 16%] Built target generate_jl_gmm_probability
[ 16%] Built target mlpack_julia_gmm_generate
[ 17%] Built target generate_jl_hmm_generate
[ 17%] Built target mlpack_julia_hmm_generate
[ 18%] Built target generate_jl_hmm_train
[ 18%] Built target mlpack_julia_hmm_loglik
[ 18%] Built target generate_jl_hmm_loglik
[ 18%] Built target mlpack_julia_hmm_viterbi
[ 18%] Built target generate_jl_hmm_viterbi
[ 19%] Built target mlpack_julia_hmm_train
[ 19%] Built target generate_jl_hoeffding_tree
[ 19%] Built target mlpack_julia_hoeffding_tree
[ 19%] Built target generate_jl_kde
[ 19%] Built target mlpack_julia_kde
[ 20%] Built target generate_jl_kernel_pca
[ 20%] Built target mlpack_julia_kernel_pca
[ 21%] Built target generate_jl_kmeans
[ 21%] Built target mlpack_julia_kmeans
[ 22%] Built target generate_jl_lars
[ 22%] Built target mlpack_julia_lars
[ 22%] Built target generate_jl_linear_regression
[ 22%] Built target mlpack_julia_linear_regression
[ 23%] Built target generate_jl_linear_svm
[ 24%] Built target mlpack_julia_linear_svm
[ 24%] Built target generate_jl_lmnn
[ 24%] Built target mlpack_julia_lmnn
[ 25%] Built target generate_jl_local_coordinate_coding
[ 25%] Built target mlpack_julia_local_coordinate_coding
[ 25%] Built target generate_jl_logistic_regression
[ 26%] Built target mlpack_julia_logistic_regression
[ 26%] Built target mlpack_julia_lsh
[ 26%] Built target generate_jl_lsh
[ 26%] Built target mlpack_julia_mean_shift
[ 27%] Built target generate_jl_mean_shift
[ 27%] Built target generate_jl_nbc
[ 28%] Built target mlpack_julia_nbc
[ 29%] Built target generate_jl_nca
[ 29%] Built target mlpack_julia_nca
[ 29%] Built target mlpack_julia_knn
[ 29%] Built target generate_jl_knn
[ 29%] Built target generate_jl_kfn
[ 30%] Built target mlpack_julia_kfn
[ 30%] Built target generate_jl_nmf
[ 30%] Built target mlpack_julia_nmf
[ 31%] Built target generate_jl_pca
[ 32%] Built target mlpack_julia_pca
[ 32%] Built target generate_jl_perceptron
[ 32%] Built target mlpack_julia_perceptron
[ 34%] Built target mlpack_julia_preprocess_describe
[ 34%] Built target generate_jl_preprocess_split
[ 35%] Built target generate_jl_preprocess_describe
[ 35%] Built target mlpack_julia_preprocess_binarize
[ 35%] Built target mlpack_julia_preprocess_split
[ 35%] Built target generate_jl_preprocess_binarize
[ 35%] Built target mlpack_julia_preprocess_one_hot_encoding
[ 35%] Built target generate_jl_preprocess_one_hot_encoding
[ 35%] Built target mlpack_julia_preprocess_scale
[ 36%] Built target generate_jl_preprocess_scale
[ 37%] Built target mlpack_julia_image_converter
[ 38%] Built target generate_jl_image_converter
[ 39%] Built target generate_jl_radical
[ 40%] Built target mlpack_julia_radical
[ 40%] Built target mlpack_julia_random_forest
[ 40%] Built target generate_jl_random_forest
[ 40%] Built target generate_jl_krann
[ 41%] Built target mlpack_julia_krann
[ 41%] Built target generate_jl_softmax_regression
[ 41%] Built target mlpack_julia_softmax_regression
[ 42%] Built target generate_jl_sparse_coding
[ 42%] Built target julia
[ 42%] Built target r_copy
[ 42%] Built target generate_r_sparse_coding
[ 43%] Built target generate_r_test_r_binding
[ 44%] Built target generate_r_adaboost
[ 44%] Built target generate_r_approx_kfn
[ 45%] Built target generate_r_bayesian_linear_regression
[ 45%] Built target generate_r_cf
[ 46%] Built target generate_r_dbscan
[ 47%] Built target generate_r_decision_tree
[ 47%] Built target generate_r_det
[ 48%] Built target generate_r_emst
[ 48%] Built target generate_r_fastmks
[ 49%] Built target generate_r_gmm_generate
[ 50%] Built target generate_r_gmm_train
[ 50%] Built target generate_r_gmm_probability
[ 50%] Built target generate_r_hmm_generate
[ 50%] Built target generate_r_hmm_train
[ 51%] Built target generate_r_hmm_loglik
[ 52%] Built target generate_r_hmm_viterbi
[ 52%] Built target generate_r_hoeffding_tree
[ 53%] Built target generate_r_kde
[ 53%] Built target generate_r_kernel_pca
[ 53%] Built target generate_r_kmeans
[ 54%] Built target generate_r_lars
[ 54%] Built target generate_r_linear_regression
[ 55%] Built target generate_r_linear_svm
[ 55%] Built target generate_r_lmnn
[ 56%] Built target generate_r_local_coordinate_coding
[ 57%] Built target generate_r_logistic_regression
[ 57%] Built target generate_r_lsh
[ 58%] Built target generate_r_mean_shift
[ 58%] Built target generate_r_nbc
[ 59%] Built target generate_r_nca
[ 60%] Built target generate_r_knn
[ 61%] Built target generate_r_kfn
[ 61%] Built target generate_r_nmf
[ 62%] Built target generate_r_pca
[ 62%] Built target generate_r_perceptron
[ 62%] Built target generate_r_preprocess_describe
[ 63%] Built target generate_r_preprocess_binarize
[ 63%] Built target generate_r_preprocess_scale
[ 64%] Built target generate_r_preprocess_one_hot_encoding
[ 65%] Built target generate_r_preprocess_split
[ 67%] Built target generate_r_image_converter
[ 68%] Built target generate_r_radical
[ 68%] Built target generate_r_random_forest
[ 68%] Built target generate_r_krann
[ 69%] Built target generate_r_softmax_regression
ℹ Loading mlpack
Error: The `pkgbuild` package is required to compile packages with a `src/` directory.
Backtrace:
    █
 1. └─roxygen2::roxygenize(".")
 2.   └─roxygen2:::load_code(base_path)
 3.     └─pkgload::load_all(path, helpers = FALSE, attach_testthat = FALSE)
 4.       └─rlang::check_installed("pkgbuild", reason = "to compile packages with a `src/` directory.")
Execution halted
make[2]: *** [src/mlpack/bindings/R/CMakeFiles/r_build.dir/build.make:78: r_build] Error 1
make[1]: *** [CMakeFiles/Makefile2:3156: src/mlpack/bindings/R/CMakeFiles/r_build.dir/all] Error 2
make: *** [Makefile:160: all] Error 2


```
But after finding the lib path via **R -e ".libPaths()"** command i pasted the  required libraries in those locations(displayed below)

**[1] "/home/devesh/R/x86_64-pc-linux-gnu-library/4.0"
[2] "/usr/local/lib/R/site-library"                 
[3] "/usr/lib/R/site-library"                       
[4] "/usr/lib/R/library"**

Now, it is displaying the error shown below, & I have no idea how to proceed further. Please help !
```
 [  0%] Updating arma_config.hpp (if necessary)
[  0%] Built target mlpack_arma_config
Moving header files to include/mlpack/
[  0%] Built target mlpack_headers
[  1%] Generating mlpack.pc (pkg-config) file.
[  1%] Built target pkgconfig
[  6%] Built target mlpack
[  6%] Built target mlpack_julia_util
[  7%] Built target mlpack_julia_sparse_coding
[  7%] Built target generate_jl_test_julia_binding
[  7%] Built target mlpack_julia_test_julia_binding
[  7%] Built target generate_jl_adaboost
[  7%] Built target mlpack_julia_adaboost
[  7%] Built target generate_jl_approx_kfn
[  7%] Built target mlpack_julia_approx_kfn
[  8%] Built target generate_jl_bayesian_linear_regression
[  9%] Built target mlpack_julia_bayesian_linear_regression
[  9%] Built target generate_jl_cf
[  9%] Built target mlpack_julia_cf
[ 10%] Built target generate_jl_dbscan
[ 10%] Built target mlpack_julia_dbscan
[ 10%] Built target generate_jl_decision_stump
[ 11%] Built target mlpack_julia_decision_stump
[ 12%] Built target generate_jl_decision_tree
[ 12%] Built target mlpack_julia_decision_tree
[ 12%] Built target generate_jl_det
[ 12%] Built target mlpack_julia_det
[ 12%] Built target generate_jl_emst
[ 12%] Built target mlpack_julia_emst
[ 13%] Built target generate_jl_fastmks
[ 14%] Built target mlpack_julia_fastmks
[ 15%] Built target mlpack_julia_gmm_train
[ 15%] Built target generate_jl_gmm_train
[ 15%] Built target generate_jl_gmm_generate
[ 15%] Built target mlpack_julia_gmm_probability
[ 16%] Built target generate_jl_gmm_probability
[ 16%] Built target mlpack_julia_gmm_generate
[ 17%] Built target generate_jl_hmm_generate
[ 17%] Built target mlpack_julia_hmm_generate
[ 18%] Built target generate_jl_hmm_train
[ 18%] Built target mlpack_julia_hmm_loglik
[ 18%] Built target generate_jl_hmm_loglik
[ 18%] Built target mlpack_julia_hmm_viterbi
[ 18%] Built target generate_jl_hmm_viterbi
[ 19%] Built target mlpack_julia_hmm_train
[ 19%] Built target generate_jl_hoeffding_tree
[ 19%] Built target mlpack_julia_hoeffding_tree
[ 19%] Built target generate_jl_kde
[ 19%] Built target mlpack_julia_kde
[ 20%] Built target generate_jl_kernel_pca
[ 20%] Built target mlpack_julia_kernel_pca
[ 21%] Built target generate_jl_kmeans
[ 21%] Built target mlpack_julia_kmeans
[ 22%] Built target generate_jl_lars
[ 22%] Built target mlpack_julia_lars
[ 22%] Built target generate_jl_linear_regression
[ 22%] Built target mlpack_julia_linear_regression
[ 23%] Built target generate_jl_linear_svm
[ 24%] Built target mlpack_julia_linear_svm
[ 24%] Built target generate_jl_lmnn
[ 24%] Built target mlpack_julia_lmnn
[ 25%] Built target generate_jl_local_coordinate_coding
[ 25%] Built target mlpack_julia_local_coordinate_coding
[ 25%] Built target generate_jl_logistic_regression
[ 26%] Built target mlpack_julia_logistic_regression
[ 26%] Built target mlpack_julia_lsh
[ 26%] Built target generate_jl_lsh
[ 26%] Built target mlpack_julia_mean_shift
[ 27%] Built target generate_jl_mean_shift
[ 27%] Built target generate_jl_nbc
[ 28%] Built target mlpack_julia_nbc
[ 29%] Built target generate_jl_nca
[ 29%] Built target mlpack_julia_nca
[ 29%] Built target mlpack_julia_knn
[ 29%] Built target generate_jl_knn
[ 29%] Built target generate_jl_kfn
[ 30%] Built target mlpack_julia_kfn
[ 30%] Built target generate_jl_nmf
[ 30%] Built target mlpack_julia_nmf
[ 31%] Built target generate_jl_pca
[ 32%] Built target mlpack_julia_pca
[ 32%] Built target generate_jl_perceptron
[ 32%] Built target mlpack_julia_perceptron
[ 34%] Built target mlpack_julia_preprocess_describe
[ 34%] Built target generate_jl_preprocess_split
[ 35%] Built target generate_jl_preprocess_describe
[ 35%] Built target mlpack_julia_preprocess_binarize
[ 35%] Built target mlpack_julia_preprocess_split
[ 35%] Built target generate_jl_preprocess_binarize
[ 35%] Built target mlpack_julia_preprocess_one_hot_encoding
[ 35%] Built target generate_jl_preprocess_one_hot_encoding
[ 35%] Built target mlpack_julia_preprocess_scale
[ 36%] Built target generate_jl_preprocess_scale
[ 37%] Built target mlpack_julia_image_converter
[ 38%] Built target generate_jl_image_converter
[ 39%] Built target generate_jl_radical
[ 40%] Built target mlpack_julia_radical
[ 40%] Built target mlpack_julia_random_forest
[ 40%] Built target generate_jl_random_forest
[ 40%] Built target generate_jl_krann
[ 41%] Built target mlpack_julia_krann
[ 41%] Built target generate_jl_softmax_regression
[ 41%] Built target mlpack_julia_softmax_regression
[ 42%] Built target generate_jl_sparse_coding
[ 42%] Built target julia
[ 42%] Built target r_copy
[ 42%] Built target generate_r_sparse_coding
[ 43%] Built target generate_r_test_r_binding
[ 44%] Built target generate_r_adaboost
[ 44%] Built target generate_r_approx_kfn
[ 45%] Built target generate_r_bayesian_linear_regression
[ 45%] Built target generate_r_cf
[ 46%] Built target generate_r_dbscan
[ 47%] Built target generate_r_decision_tree
[ 47%] Built target generate_r_det
[ 48%] Built target generate_r_emst
[ 48%] Built target generate_r_fastmks
[ 49%] Built target generate_r_gmm_generate
[ 50%] Built target generate_r_gmm_train
[ 50%] Built target generate_r_gmm_probability
[ 50%] Built target generate_r_hmm_generate
[ 50%] Built target generate_r_hmm_train
[ 51%] Built target generate_r_hmm_loglik
[ 52%] Built target generate_r_hmm_viterbi
[ 52%] Built target generate_r_hoeffding_tree
[ 53%] Built target generate_r_kde
[ 53%] Built target generate_r_kernel_pca
[ 53%] Built target generate_r_kmeans
[ 54%] Built target generate_r_lars
[ 54%] Built target generate_r_linear_regression
[ 55%] Built target generate_r_linear_svm
[ 55%] Built target generate_r_lmnn
[ 56%] Built target generate_r_local_coordinate_coding
[ 57%] Built target generate_r_logistic_regression
[ 57%] Built target generate_r_lsh
[ 58%] Built target generate_r_mean_shift
[ 58%] Built target generate_r_nbc
[ 59%] Built target generate_r_nca
[ 60%] Built target generate_r_knn
[ 61%] Built target generate_r_kfn
[ 61%] Built target generate_r_nmf
[ 62%] Built target generate_r_pca
[ 62%] Built target generate_r_perceptron
[ 62%] Built target generate_r_preprocess_describe
[ 63%] Built target generate_r_preprocess_binarize
[ 63%] Built target generate_r_preprocess_scale
[ 64%] Built target generate_r_preprocess_one_hot_encoding
[ 65%] Built target generate_r_preprocess_split
[ 67%] Built target generate_r_image_converter
[ 68%] Built target generate_r_radical
[ 68%] Built target generate_r_random_forest
[ 68%] Built target generate_r_krann
[ 69%] Built target generate_r_softmax_regression
ℹ Loading mlpack
Error: The `pkgbuild` package is required to compile packages with a `src/` directory.
Backtrace:
    █
 1. └─roxygen2::roxygenize(".")
 2.   └─roxygen2:::load_code(base_path)
 3.     └─pkgload::load_all(path, helpers = FALSE, attach_testthat = FALSE)
 4.       └─rlang::check_installed("pkgbuild", reason = "to compile packages with a `src/` directory.")
Execution halted
make[2]: *** [src/mlpack/bindings/R/CMakeFiles/r_build.dir/build.make:78: r_build] Error 1
make[1]: *** [CMakeFiles/Makefile2:3156: src/mlpack/bindings/R/CMakeFiles/r_build.dir/all] Error 2
make: *** [Makefile:160: all] Error 2

```

**Comment by rcurtin** [comment]
Hey @Kashyapdevesh, I took a look into the issue.  I think you can solve it by running `install.packages("pkgbuild")` or otherwise installing `pkgbuild` from CRAN.

However, definitely our build system should tell you about the missing requirement at compile time!  So, I put together a PR that should fix the issue moving forwards: #3081.

**Comment by coatless** [comment]
The patch @rcurtin should fix this build issue. We just were missing out on a development package being present that usually would be avoided if we were not regenerating the package in its entirety.

**Comment by Kashyapdevesh** [comment]
I had already individually installed pkgbuild when I was facing the issue of missing packages. I think there was some issue with correct file paths and R bindings, as previously I had installed R using conda and the mlpack was not syncing to the correct file path. Nevertheless, I deleted conda dependencies, made a global installation of R, and rebuilt mlpack and it worked.
Btw, @rcurtin thanks for the patch and for looking after the issue.
